### PR TITLE
Add recovery code as possible requestAuth method.

### PR DESCRIPTION
**Pull request by joseluisdiaz**
This allow to use the recovery code directly when is the only Authenticator.

## Depends one 
https://github.com/auth0/auth0-mfa-api/pull/724

## Dependent
https://github.com/auth0/mfa-widget/pull/99

### CODE REVIEW

**Comment by dafortune** [comment]
@joseluisdiaz Nothing to update in the readme?

**Comment by joseluisdiaz** [comment]
Good point; adding the required payload when use `requestAuth`

File: lib/transaction/index.js [hunk 1]
```diff
@@ -162,7 +163,7 @@ transaction.prototype = object.create(EventEmitter.prototype);
 
 /**
  * @param {object} data
- * @param {otp|push|sms} data.method
+ * @param {otp|push|sms|recovery-code} data.method
```
Comment (dafortune) [inline lib/transaction/index.js]: Is method going to include recovery code? doesnt that cause the issue we talked about?

**Comment by dafortune** [comment]
Let's check https://github.com/auth0/auth0-mfa-api/pull/680#issuecomment-364029198 before merging this. Also I think there is a missing point in the readme

**Comment by dafortune** [comment]
Dear reviewer, I've added a new method getAvailableAuthenticatorTypes to mimic the api changes, however, maybe we can just change the behavior of getAvailableMethods to return that field instead of the current one and consider it a breaking change? (We cannot change the behavior of the server to return the new value as part of the "availableMethods" because that would break current users that get trapped in the "only recovery code" case because the current version of the library does not support recovery code.

**Comment by dafortune** [comment]
Closing in favor of https://github.com/auth0/auth0-guardian.js/pull/47

**Pull request closed without merging**